wetterdienst/cli.py

```diff
@@ -294,44 +302,64 @@ def run():
                 longitude=float(options.longitude),
                 max_distance_in_km=int(options.distance),
             )
+        else:
+            raise DocoptExit("Geospatial queries need either --number or --distance")
+        results = stations
+
     elif options.station:
-        stations = stations.filter(read_list(options.station))
+        results = stations.filter(read_list(options.station))
+
     else:
-        stations = stations.all()
+        results = stations.all()
 
     df = pd.DataFrame()
 
     if options.stations:
-        df = stations.df
+        pass
+
     elif options["values"]:
         try:
             # TODO: Add stream-based processing here.
-            df = stations.values.all().df
+            results = results.values.all()
         except ValueError as ex:
             log.exception(ex)
             sys.exit(1)
 
+    df = results.df
+
     if df.empty:
         log.error("No data available for given constraints")
         sys.exit(1)
 
     # Filter readings by datetime expression.
     if options["values"] and options.date:
-        df = df.dwd.filter_by_date(options.date, stations.stations.resolution)
+        results.filter_by_date(options.date)
 
     # Apply filtering by SQL.
     if options.sql:
+        if options.tidy:
+            log.error("Combining SQL filtering with tidy format not possible")
+            sys.exit(1)
+
         log.info(f"Filtering with SQL: {options.sql}")
-        df = df.io.sql(options.sql)
+        results.filter_by_sql(options.sql)
 
     # Emit to data sink, e.g. write to database.
     if options.target:
-        df.io.export(options.target)
+        results.to_target(options.target)
         return
 
     # Render to output format.
     try:
-        output = df.dwd.format(options.format)
+        if options.format == "json":
```

Comment:
maybe we could use **results.export(fmt)** here

Reply:
Yes, we can well move this method. However, it is more a formatter than an exporter. Export stuff is in `ExportMixin.to_target()`.